src/graph/service/PermissionCheck.cpp

```diff
@@ -109,8 +109,19 @@ namespace graph {
     case Sentence::Kind::kUpdateEdge:
     case Sentence::Kind::kDeleteVertices:
     case Sentence::Kind::kDeleteTags:
-    case Sentence::Kind::kDeleteEdges:
+    case Sentence::Kind::kDeleteEdges: {
+      return PermissionManager::canWriteData(session, vctx);
+    }
     case Sentence::Kind::kAdminJob: {
+      auto *adminJobSentence = dynamic_cast<AdminJobSentence *>(sentence);
+      if (adminJobSentence == nullptr) {
+        // should not happend.
+        LOG(WARNING) << "sentence is not AdminJobSentence";
+        return Status::PermissionError("No permission to write space.");
```

Comment:
should the message be sth like "bad format sentence"?

Reply:
> should the message be sth like "bad format sentence"?

It's a path should never go in, I wan't use check not null first...  Make sence to me.